src/Functions/DivisionUtils.h

```diff
@@ -86,8 +87,15 @@ struct ModuloImpl
     template <typename Result = ResultType>
     static inline Result apply(A a, B b)
     {
-        throwIfDivisionLeadsToFPE(typename NumberTraits::ToInteger<A>::Type(a), typename NumberTraits::ToInteger<B>::Type(b));
-        return typename NumberTraits::ToInteger<A>::Type(a) % typename NumberTraits::ToInteger<B>::Type(b);
+        if constexpr (std::is_floating_point_v<ResultType>)
+        {
+            return ResultType(a) - trunc(ResultType(a) / ResultType(b)) * ResultType(b);
```

Comment:
why not `fmod`?

Reply:
I did not aware of this function (what a shame), let's test it...